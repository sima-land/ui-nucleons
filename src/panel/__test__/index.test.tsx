import { render } from '@testing-library/react';
import { Panel } from '..';
import LockSVG from '@sima-land/ui-quarks/icons/16x16/Stroked/Lock';
import InformationSVG from '@sima-land/ui-quarks/icons/16x16/Stroked/Information';

describe('<Panel />', () => {
  it('should renders without props', () => {
    const { container } = render(<Panel />);

    expect(container).toMatchSnapshot();
  });

  it('should handle props', () => {
    const { container } = render(
      <Panel adornmentStart={<InformationSVG />} adornmentEnd={<LockSVG />} className='test-class'>
        Hello, world!
      </Panel>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should handle unknown content', () => {
    const markup = 'Lorem <i data-testid="ipsum">ipsum</i> <b>dolor</b> sit.';

    const { container, queryAllByTestId } = render(
      <Panel className='test-class'>
        <Panel.UnknownContent markup={markup} />
      </Panel>,
    );

    expect(container).toMatchSnapshot();
    expect(queryAllByTestId('ipsum')).toHaveLength(1);
  });

  it('should handle "data-testid" prop', () => {
    const { container } = render(<Panel data-testid='other-panel'>Hello, world!</Panel>);

    expect(container.querySelectorAll('[data-testid="modal"]')).toHaveLength(0);
    expect(container.querySelectorAll('[data-testid="other-panel"]')).toHaveLength(1);
  });
});
