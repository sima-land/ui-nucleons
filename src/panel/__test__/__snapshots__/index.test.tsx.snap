// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should handle props 1`] = `
<div>
  <div
    class="root type-info bg-color__basic-gray4 color__basic-gray87 test-class"
    data-testid="panel"
  >
    <div
      class="adornment"
    >
      <svg
        height="16"
        viewBox="0 0 16 16"
        width="16"
      >
        <path
          d="M8 6.5a1 1 0 0 1 1 1v4a1 1 0 1 1-2 0v-4a1 1 0 0 1 1-1ZM8 5.5a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z"
        />
        <path
          clip-rule="evenodd"
          d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6a6 6 0 1 0 0 12A6 6 0 0 0 8 2Z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="main"
    >
      Hello, world!
    </div>
    <div
      class="adornment"
    >
      <svg
        height="16"
        viewBox="0 0 16 16"
        width="16"
      >
        <path
          d="M9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z"
        />
        <path
          clip-rule="evenodd"
          d="M4 6V4a4 4 0 1 1 8 0v2a3 3 0 0 1 3 3v4a3 3 0 0 1-3 3H4a3 3 0 0 1-3-3V9a3 3 0 0 1 3-3Zm2-2a2 2 0 1 1 4 0v2H6V4ZM4 8a1 1 0 0 0-1 1v4a1 1 0 0 0 1 1h8a1 1 0 0 0 1-1V9a1 1 0 0 0-1-1H4Z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Panel should handle unknown content 1`] = `
<div>
  <div
    class="root type-info bg-color__basic-gray4 color__basic-gray87 test-class"
    data-testid="panel"
  >
    <div
      class="main"
    >
      <div
        class="unknown-content"
      >
        Lorem 
        <i
          data-testid="ipsum"
        >
          ipsum
        </i>
         
        <b>
          dolor
        </b>
         sit.
      </div>
    </div>
  </div>
</div>
`;

exports[`Panel should renders without props 1`] = `
<div>
  <div
    class="root type-info bg-color__basic-gray4 color__basic-gray87"
    data-testid="panel"
  >
    <div
      class="main"
    />
  </div>
</div>
`;
